sdk/core/Azure.Core/src/Pipeline/HttpPipeline.cs

```diff
@@ -144,6 +199,51 @@ public static IDisposable CreateHttpMessagePropertiesScope(IDictionary<string, o
             return CurrentHttpMessagePropertiesScope.Value;
         }
 
+        private ReadOnlyMemory<HttpPipelinePolicy> CreateRequestPipeline(HttpPipelinePolicy[] policies, List<(HttpPipelinePosition Position, HttpPipelinePolicy Policy)> customPolicies)
```

Comment:
This seems like an expensive routine to execute on every call. @pakrym stipulated that we can implement it as a sequence of 5 (I think) array copy operations.

Reply:
I think it's seven... I'll do another pass over it.  Thanks!

Reply:
I think it would need to be seven.  But yes, happy to revise here to improve efficiency - thanks!

Reply:
@KrzysztofCwalina, revised and thank you for your help on this!  Let me know if any further improvements are needed to this routine.